src/serialize.h

```diff
@@ -520,6 +520,15 @@ struct VarIntFormatter
     }
 };
 
+/** Serialization wrapper class for custom integers and enums.
+ *
+ * It permits specifying the serialized size (1 to 8 bytes) and endianness.
+ *
+ * Use the big endian mode for values that are stored in memory in native
+ * byte order, but serialized in big endian notation. This is only intended
+ * to implement serializers that are compatible with existing formats, and
+ * its use is not recommended for new data structures.
```

Comment:
in commit f9ee0f37c28f604bc82dab502ce229c66ef5b3b9

Not recommend for big endian or anything that can be serialized with this wrapper? For example, I am wondering if there is a recommended way to serialize enums or should new code rather not use enums instead?

Reply:
The comment is only intended to apply to big endian serialization. Any suggestions to make that clearer?

If it's confusing, I think it's also fine to just drop the comment. When picking a serialization for a new structure I doubt anyone would intentionally complicate their code by using big endian serialization.

Reply:
> I doubt anyone would intentionally complicate their code by using big endian serialization

Makes sense

Reply:
In commit "Add comments to CustomUintFormatter" (f9ee0f37c28f604bc82dab502ce229c66ef5b3b9)

re: "not recommended for new data structures", it'd be nice to say what preferred alternatives are. Compact ints, varints, little endian fixed width ints? I keep forgetting all the int types myself..

Reply:
@ryanofsky Maybe the serialization code just shouldn't contain protocol design advice.